src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -17,18 +24,42 @@ export interface ThemeProviderProps {
    * String which adds up to theme name selector to make it more specific (in case if theme.name is colliding with some other class name)
    */
   themeClassSpecifier?: string;
+  /**
+   * The system theme to apply to the body element on mount, if there is no theme class name on the body element already
+   */
+  systemTheme?: SystemTheme;
 }
 
 const ThemeProvider: FC<ThemeProviderProps> & {
   systemThemes?: typeof SystemTheme;
   colors?: typeof ThemeColor;
-} = ({ theme, children, themeClassSpecifier: customThemeClassSpecifier }) => {
+} = ({ theme, children, themeClassSpecifier: customThemeClassSpecifier, systemTheme }) => {
   const [stylesLoaded, setStylesLoaded] = useState(false);
   const themeClassSpecifier = useMemo(
     () => customThemeClassSpecifier || generateRandomAlphaString(),
     [customThemeClassSpecifier]
   );
 
+  // Add the systemTheme class name to the body on mount
+  useLayoutEffect(() => {
```

Comment:
I'm missing a test for this logic.

It can be either an interaction test or a dedicated hook testing